File: app/src/main/java/com/rdev/tryp/createAccount/SignUpFragment.java
package com.rdev.tryp.createAccount;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.ImageButton;

import com.rdev.tryp.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.cardview.widget.CardView;
import androidx.fragment.app.Fragment;

public class SignUpFragment extends Fragment implements View.OnClickListener {
    ImageButton back_btn;
    CardView sign_up_card;
    EditText sign_up_first_name_et;
    EditText sign_up_last_name_et;
    EditText sign_up_email_et;
    EditText sign_up_reffered_et;


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.sign_up_fragment, container, false);
        back_btn = v.findViewById(R.id.back_btn);
        back_btn.setOnClickListener(this);
        sign_up_card = v.findViewById(R.id.sign_up_card);
        sign_up_card.setOnClickListener(this);
        sign_up_first_name_et = v.findViewById(R.id.sign_up_first_name_et);
        sign_up_last_name_et = v.findViewById(R.id.sign_up_last_name_et);
        sign_up_email_et = v.findViewById(R.id.sign_up_email_et);
        sign_up_reffered_et = v.findViewById(R.id.sign_up_reffered_et);
        return v;
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.back_btn:
                ((CreateActivity) getActivity()).goBack();
                break;
            case R.id.sign_up_card:
                ((CreateActivity) getActivity()).createUser.setLast_name(sign_up_last_name_et.getText().toString());
                ((CreateActivity) getActivity()).createUser.setFirst_name(sign_up_first_name_et.getText().toString());
                ((CreateActivity) getActivity()).createUser.setEmail(sign_up_email_et.getText().toString());
                ((CreateActivity) getActivity()).createUser.setRef_code(sign_up_reffered_et.getText().toString());
                ((CreateActivity) getActivity()).signUp();
                break;

        }
    }
}

File: app/src/main/java/com/rdev/tryp/createAccount/SignUpFragment.kt
package com.rdev.tryp.createAccount

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.ImageButton

import com.rdev.tryp.R
import com.rdev.tryp.intro.createAccount.CreateActivity
import androidx.cardview.widget.CardView
import androidx.fragment.app.Fragment


class SignUpFragment : Fragment(), View.OnClickListener {

    private lateinit var btnBack: ImageButton
    private lateinit var signUpCard: CardView
    private lateinit var etFirstName: EditText
    private lateinit var etLastName: EditText
    private lateinit var etEmail: EditText
    private lateinit var etInvitedBy: EditText

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.sign_up_fragment, container, false)

        btnBack = v.findViewById(R.id.back_btn)
        btnBack.setOnClickListener(this)
        signUpCard = v.findViewById(R.id.sign_up_card)
        signUpCard.setOnClickListener(this)
        etFirstName = v.findViewById(R.id.sign_up_first_name_et)
        etLastName = v.findViewById(R.id.sign_up_last_name_et)
        etEmail = v.findViewById(R.id.sign_up_email_et)
        etInvitedBy = v.findViewById(R.id.sign_up_reffered_et)

        return v
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.back_btn       -> (activity as CreateActivity).goBack()
            R.id.sign_up_card   -> {
                (activity as? CreateActivity)?.createUser?.last_name    =   etLastName.text.toString()
                (activity as? CreateActivity)?.createUser?.first_name   =   etFirstName.text.toString()
                (activity as? CreateActivity)?.createUser?.email        =   etEmail.text.toString()
                (activity as? CreateActivity)?.createUser?.ref_code     =   etInvitedBy.text.toString()
                (activity as? CreateActivity)?.signUp()
            }
        }
    }

}